src/githubApi.ts

```diff
@@ -0,0 +1,33 @@
+import { GitHub } from "@actions/github/lib/utils";
+import { PullRequest, Repo } from "./types";
+import { debug } from "@actions/core";
+
+const listPrs = (octokit: InstanceType<typeof GitHub>, repo: Repo, per_page: number = 100, page: number = 1) => {
+    return octokit.rest.pulls.list({ ...repo, per_page, state: "open", page });
+}
+
+export const getPullRequestWithReviews = async (octokit: InstanceType<typeof GitHub>, repo: Repo):Promise<PullRequest[]> => {
+    const perPage = 100;
+    let currentPage = 1;
+    const { data } = await listPrs(octokit, repo, perPage, currentPage);
+    let prs = data;
+    let fullPage = prs.length > 99;
+    while (fullPage) {
+        currentPage++;
+        debug(`Iterating on page ${currentPage} with ${prs.length} issues`);
+        const { data } = await listPrs(octokit, repo, perPage, currentPage);
+        prs = prs.concat(data);
+        fullPage = data.length > 99;
+    }
```

Comment:
I'm gonna keep reminding you that we have this logic already implemented in `@eng-automation/integrations` :)

Here's an example usage:
https://github.com/paritytech/gitspiegel/blob/c2835ece6cc230d62e23fbdf1987313a122e86f8/src/core.ts#L181

Also, in case of GHA, even passing an octokit instance there shouldn't be required, as `GITHUB_TOKEN` is already provided

Reply:
Implemented! Thank you for your help!